cache/src/test/kotlin/com/dropbox/android/external/cache4/CacheLoaderTest.kt

```diff
@@ -276,42 +246,36 @@ class CacheLoaderTest {
 
             val loader = createSlowLoader("cat", executionTime)
 
-            repeat(3) {
+            (0 until 3).map {
                 launch(newSingleThreadDispatcher()) {
                     // each call uses a different key
                     cache.get(it.toLong(), loader)
                 }
-            }
-
-            delay(executionTime * 3 + 10)
+            }.joinAll()
 
-            assertThat(loader.invokeCount)
-                .isEqualTo(3)
+            assertEquals(3, loader.invokeCount)
 
-            assertThat(cache.get(1))
-                .isEqualTo("cat")
+            assertEquals("cat", cache.get(1))
         }
 
     @Test
     fun `a loader exception is propagated to the get(key, loader) call`() {
         val cache = Cache.Builder.newBuilder()
             .build<Long, String>()
 
-        val loader = createFailingLoader(IOException())
+        val loader = createFailingLoader(RuntimeException())
```

Comment:
I generally prefer to throw more specific exceptions and **detekt** would consider `RuntimeException` to be too specific.

Reply:
We can create a very specific exception just for this test, like SomeLoaderException()